graphql/src/store/resolver.rs

```diff
@@ -164,62 +163,22 @@ where
     /// Resolves a scalar value for a given scalar type.
     fn resolve_scalar_value(
         &self,
-        parent_object_type: &s::ObjectType,
-        parent: &BTreeMap<String, q::Value>,
+        _parent_object_type: &s::ObjectType,
+        _parent: &BTreeMap<String, q::Value>,
         field: &q::Field,
         scalar_type: &s::ScalarType,
         value: Option<&q::Value>,
     ) -> Result<q::Value, QueryExecutionError> {
-        let subgraph_id = parse_subgraph_id(parent_object_type).unwrap();
-
-        // Extract __typename from the parent object
-        let typename = parent
-            .get("__typename")
-            .and_then(|typename| match typename {
-                q::Value::String(typename) => Some(typename),
-                _ => None,
-            })
-            .expect("GraphQL object must have `__typename`");
-
-        match (
-            subgraph_id.deref().as_str(),
-            typename.as_str(),
-            field.name.as_str(),
-        ) {
-            // Compute `entityCount` on-demand
-            ("subgraphs", "SubgraphDeployment", "entityCount") => {
-                // Extract the entity ID
-                let id = parent
-                    .get("id")
-                    .and_then(|id| match id {
-                        q::Value::String(id) => Some(id),
-                        _ => None,
-                    })
-                    .expect("GraphQL object must have an `id`");
-
-                // Convert the ID to a subgraph deployment ID
-                let hash =
-                    SubgraphDeploymentId::new(id.clone()).expect("invalid subgraph ID in database");
-
-                // Return the entity count of the deployment as a BigInt
-                Ok(Value::from(
-                    self.store
-                        .count_entities(hash)
-                        .map_err(QueryExecutionError::StoreError)?,
+        value.map_or(Ok(q::Value::Null), |value| {
```

Comment:
We could drop the entire method now, since the `Resolver` trait has a default implementation that probably does the same. Worth checking.

Reply:
Yep, you are right - the two were character-for-character identical